### PR TITLE
#36636_index_create_category

### PR DESCRIPTION
**Pull request by trinhducduy**

### CODE REVIEW

File: app/controllers/admin/base_controller.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class Admin::BaseController < ApplicationController
+
```
Comment (ThuBM) [inline app/controllers/admin/base_controller.rb]: remove

File: app/views/admin/categories/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+    <table class="table table-striped">
+      <tr>
+        <th><%= t ".name" %></th>
+        <th><%= t ".description" %></th>
+        <th></th>
+        <th></th>
+      </tr>
+
+      <% @categories.each do |category| %>
```
Comment (ThuBM) [inline app/views/admin/categories/index.html.erb]: write in _category.html.erb

File: app/controllers/admin/categories_controller.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
+class Admin::CategoriesController < Admin::BaseController
+  def index
+    @categories = Category.latest.paginate page: params[:page],
+      per_page: 10
```
Comment (ThuBM) [inline app/controllers/admin/categories_controller.rb]: 10 -> Settings

File: app/views/admin/categories/index.html.erb [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+          class: "btn btn-primary" %>
+      </span>
+    </h1>
+
+    <table class="table table-striped">
+      <tr>
+        <th><%= t ".name" %></th>
+        <th><%= t ".description" %></th>
+        <th></th>
+        <th></th>
```
Comment (HieuMeo) [inline app/views/admin/categories/index.html.erb]: ?

File: app/controllers/admin/categories_controller.rb [hunk 5]
```diff
@@ -0,0 +1,26 @@
+class Admin::CategoriesController < Admin::BaseController
+  def index
+    @categories = Category.latest.paginate page: params[:page],
+      per_page: Rails.configuration.x.pagination.page_size
```
Comment (HieuMeo) [inline app/controllers/admin/categories_controller.rb]: Use rails-settings

File: app/controllers/admin/categories_controller.rb [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+    @category = Category.new
+  end
+
+  def create
+    @category = Category.new category_params
+    if @category.save
+      flash[:success] = t "application.flash.category_created"
+      redirect_to admin_categories_path
+    else
+      flash[:danger] = t "application.flash.category_created_failed"
```
Comment (dieunb) [inline app/controllers/admin/categories_controller.rb]: no need

File: config/application.rb [hunk 7]
```diff
@@ -17,19 +17,7 @@
...
-    # Run "rake -D time" for a list of tasks for finding time zone names. Default is UTC.
-    # config.time_zone = 'Central Time (US & Canada)'
-
-    # The default locale is :en and all translations from config/locales/*.rb,yml are auto loaded.
-    # config.i18n.load_path += Dir[Rails.root.join('my', 'locales', '*.{rb,yml}').to_s]
-    # config.i18n.default_locale = :de
-
-    # Do not swallow errors in after_commit/after_rollback callbacks.
     config.active_record.raise_in_transactional_callbacks = true
+    config.x.pagination.page_size = 1
```
Comment (dieunb) [inline config/application.rb]: move to Setting file

File: config/routes.rb [hunk 8]
```diff
@@ -1,4 +1,8 @@
 Rails.application.routes.draw do
   root "books#index"
   get "about", to: "static_pages#about"
+
+  namespace :admin do
+    resources :categories
```
Comment (dieunb) [inline config/routes.rb]: root page

**Comment by dieunb** [comment]
OK

**Pull request merged**